interfaces/java/javaSwigGenerateOnly.cmake

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright (c) 2017-2019, Battelle Memorial Institute; Lawrence Livermore
+# National Security, LLC; Alliance for Sustainable Energy, LLC.
+# See the top-level NOTICE for additional details. 
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+#File used to generate the matlab interface and overwrite the existing files if requested
+
+file(GLOB SHARED_LIB_HEADERS ${CMAKE_SOURCE_DIR}/src/helics/shared_api_library/*.h)
+
+  # custom command for building the wrap file
+  add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/helicsJava.c
+    COMMAND
+      "${SWIG_EXECUTABLE}" "-java" "-package" "com.java.helics" -o "helicsJava.c"
```

Comment:
I agree with you,  this would probably be a breaking change however  I am not sure how big a deal that would be.  Definitely HELICS 3.0 should make the change, but not sure if that is something we can change right now.

Reply:
Knowing how many users there are for the interfaces would be useful. I'd guess Java is less used than the C++ and Python interfaces.

Reply:
Might have to ask @afisher1  about that.